src/state/nftMarket/types.ts

```diff
@@ -9,26 +9,29 @@ export interface State {
     collections: Record<string, Collection> // string is the address
     nfts: Record<string, NFT[]> // string is the address
     users: Record<string, User> // string is the address
+    userNfts: NftSubgraphEntity[]
   }
 }
 
 export interface Transaction {
-  hash: string
+  id: string
   block: BigNumberish
   timestamp: BigNumberish
-  price: BigNumberish
+  askPrice: BigNumberish
   netPrice: BigNumberish
-  buyerAddress: string
-  sellerAddress: string
+  buyer: { id: string }
+  seller: { id: string }
   withBNB: boolean
 }
 
-export interface NFT extends NftMeta {
+export interface NftSubgraphEntity {
   tokenId: BigNumberish
-  metadataUrl: string
-  transactionHistory: Transaction[]
+  metadataUrl?: string
+  transactionHistory?: Transaction[]
```

Comment:
Making these optional as NFTs that have not been sold before will not be indexed and therefore not have either of these data points